prow/jira/fakejira/fake_test.go

```diff
@@ -72,3 +78,203 @@ func TestFakeClient_SearchWithContext(t *testing.T) {
 		t.Fatal("expected invalid query to fail, but got no error")
 	}
 }
+
+func TestCreateIssue(t *testing.T) {
+	testCases := []struct {
+		name             string
+		issue            jira.Issue
+		existingIssues   []jira.Issue
+		expectedIssue    *jira.Issue
+		createIssueError *jiraclient.CreateIssueError
+	}{{
+		name: "create single issue",
+		issue: jira.Issue{
+			Fields: &jira.IssueFields{
+				Project: jira.Project{
+					Name: "ABC",
+				},
+			},
+		},
+		expectedIssue: &jira.Issue{
+			ID:  "1",
+			Key: "ABC-1",
+			Fields: &jira.IssueFields{
+				Project: jira.Project{
+					Name: "ABC",
+				},
+			},
+		},
+	}, {
+		name: "create issue with other issues in other projects",
+		issue: jira.Issue{
+			Fields: &jira.IssueFields{
+				Project: jira.Project{
+					Name: "ABC",
+				},
+			},
+		},
+		existingIssues: []jira.Issue{
+			{
+				ID:  "22",
+				Key: "ABC-41",
+				Fields: &jira.IssueFields{
+					Project: jira.Project{
+						Name: "ABC",
+					},
+				},
+			},
+			{
+				ID:  "52",
+				Key: "DEF-16",
+				Fields: &jira.IssueFields{
+					Project: jira.Project{
+						Name: "DEF",
+					},
+				},
+			},
+		},
+		expectedIssue: &jira.Issue{
+			ID:  "53",
+			Key: "ABC-42",
+			Fields: &jira.IssueFields{
+				Project: jira.Project{
+					Name: "ABC",
+				},
+			},
+		},
+	}, {
+		name: "create issue with comments and Status set",
+		issue: jira.Issue{
+			Fields: &jira.IssueFields{
+				Project: jira.Project{
+					Name: "ABC",
+				},
+				Status: &jira.Status{
+					Name: "NEW",
+				},
+				Comments: &jira.Comments{
+					Comments: []*jira.Comment{{Body: "Hello"}},
+				},
+			},
+		},
+		createIssueError: &jiraclient.CreateIssueError{
+			Errors: map[string]string{
+				"comment": "this field cannot be set",
+				"status":  "this field cannot be set",
+			},
+		},
+	}}
+	for _, tc := range testCases {
+		ptrIssues := []*jira.Issue{}
+		for index := range tc.existingIssues {
+			ptrIssues = append(ptrIssues, &tc.existingIssues[index])
+		}
+		fakeClient := &FakeClient{Issues: ptrIssues}
+		newIssue, err := fakeClient.CreateIssue(&tc.issue)
```

Comment:
These are unit tests for a fake ... ? This doesn't seem right - in the fake you might inject data, or validate that calls happen to the right endpoints. There's nothing making sure that the fake implementation of the clone here is the same as what Jira actually does, so ensuring it is "correct" here is not helpful.

Reply:
I have removed the unit tests.